spring-webflux/src/test/java/org/springframework/web/reactive/filter/HiddenHttpMethodFilterTests.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.web.reactive.filter;
+
+import org.junit.Test;
+import org.springframework.http.HttpMethod;
+import org.springframework.mock.http.server.reactive.test.MockServerHttpRequest;
+import org.springframework.mock.http.server.reactive.test.MockServerHttpResponse;
+import org.springframework.web.server.ServerWebExchange;
+import org.springframework.web.server.WebFilterChain;
+import org.springframework.web.server.adapter.DefaultServerWebExchange;
+import reactor.core.publisher.Mono;
+
+import static org.junit.Assert.*;
+
+/**
+ * @author Greg Turnquist
+ */
+public class HiddenHttpMethodFilterTests {
+
+	private final HiddenHttpMethodFilter filter = new HiddenHttpMethodFilter();
+
+	@Test
+	public void filterWithParameter() {
+		ServerWebExchange mockExchange = createExchange();
+		mockExchange.getAttributes().put("_method", "DELETE");
+
+		WebFilterChain filterChain = exchange -> {
+			assertEquals("Invalid method", HttpMethod.DELETE, exchange.getRequest().getMethod());
+			return Mono.empty();
+		};
+		filter.filter(mockExchange, filterChain);
```

Comment:
If you don't `.block()` here, or use `StepVerifier`, your tests may not run entirely or may exit before execution is finished. It works here because we're always returning `Mono.empty()`, but this means those `Mono` aren't subscribed to and that the chain execution is not done.